modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceRaftGroupEventsListener.java

```diff
@@ -181,19 +180,9 @@ public void onNewPeersConfigurationApplied(PeersAndLearners configuration, long
         }
 
         try {
-            rebalanceScheduler.schedule(() -> {
```

Comment:
In what thread will we do this now? Please pay attention, that we should not call doStableKeySwitch within raft thread.

Reply:
Why shouldn't we? It does not block anymore

Reply:
Simply speaking, it's fragile and brings encapsulation leaks. It does not block anymore - for sure someone may add leave-weight processing here and will never detect that it's performed within raft thread. Same for exception handling, etc.

Reply:
Restored resubmission to the pool